src/gluonts/support/autogluont_adapter.py

```diff
@@ -0,0 +1,96 @@
+
+    @staticmethod
+    def get_ds(dataset_name):
+        return get_dataset(dataset_name, regenerate=False)
+
+    def conversion(self):
+        train_series, test_series = to_pandas(next(iter(self.dataset.train))), to_pandas(
+            next(iter(self.dataset.test)))
+        train_series = self.sample_train(train_series)
+        train_df, test_df = self.parse_series(train_series), self.parse_series(test_series)
+        return task.Dataset(train_df), task.Dataset(test_df)
+        # train_series1, train_series2 = self.split_train(train_series)
+        # train_df1, train_df2, test_df = self.parse_series(train_series1), self.parse_series(train_series2), self.parse_series(test_series)
+        # return task.Dataset(train_df1), task.Dataset(train_df2), task.Dataset(test_df)
+
+    def sample_train(self, series):
+        l = - (-series.size // 5)
+        new_series = pd.core.series.Series(dtype='float64')
+        for i in range(l-1):
+            subseries = series[i*5:(i+1)*5].sample(1)
+            new_series = new_series.append(subseries)
+        return new_series
+
+# get a random data point from each 5 points to reduce overfitting.
+    def split_train(self, series):
+        mini, maxi = series.values.min(), series.values.max()
+        splitter = (mini+maxi) // 2
+        train1, train2 = series.loc[lambda series: series < splitter], series.loc[lambda series: series >= splitter]
+        print(train1.head(10), train2.head(10))
+        # train1.plot()
+        # plt.grid(which="both")
+        # plt.legend(["train series"], loc="upper left")
+        # plt.title("The smaller set")
+        # plt.show()
+        #
+        # train2.plot()
+        # plt.grid(which="both")
+        # plt.legend(["train series"], loc="upper left")
+        # plt.title("The larger set")
+        # plt.show()
+
+        return train1, train2
+
+    @staticmethod
+    def parse_series(series):
+        hour_of_day = series.index.hour
+        month_of_year = series.index.month
+        day_of_week = series.index.dayofweek
+        year_idx = series.index.year
+        week_of_year = series.index.week
+        target = series.values
+        cal = calendar()
+        holidays = cal.holidays(start=series.index.min(), end=series.index.max())
+        df = pd.DataFrame(zip(year_idx, month_of_year, day_of_week, hour_of_day, series.index.isin(holidays), target),
+                         columns=['year_idx', 'month_of_year', 'day_of_week', 'hour_of_day', 'holiday', 'target'])
+        convert_type = {x: 'category' for x in df.columns.values[:-2]}
+        df = df.astype(convert_type)
+        return df
+
+if __name__ == '__main__':
+    g = Gluonts2Auto('electricity')
+    # train_data1, train_data2, test_data = g.conversion()
+    train_data, test_data = g.conversion()
+    label_column = 'target'
+    predictor = task.fit(train_data=train_data, label=label_column, problem_type='regression', output_directory='AutogluonModels/ag-a')
```

Comment:
You are not separating the train/val data by time, so it will be randomly selected. This is a problem for time-series data.

Additionally, you should try something like hyperparameters={'LR': {}} as a test, since linear regression can extrapolate, but most of AG's models cannot.

Finally, if you suspect the error is still unreasonable, try fitting a linear regression model or some kind of classical model outside of AutoGluon and comparing.

Reply:
Also, to confirm that the data is being processed by AutoGluon's feature generator correctly, please also call predictor.transform_features() after fitting to see the input data that the ML models receive.

Reply:
Hi Nick. Thanks for the comment! 
"You are not separating the train/val data by time, so it will be randomly selected. This is a problem for time-series data."
I think I got the data by time since they derive from dataset.train, dataset.test in the Gluonts model. Are you looking at the function "split_train", which is the one that planned for incremental training but never used?

Reply:
Inside of task.fit, train_data is split into train and validation for model training with early stopping. You need to instead pass train_data=train_data, tuning_data=validation_data in task.fit by first splitting train_data into train_data and validation_data where validation_data is later in time.

Reply:
Tons of thanks!